### PR TITLE
Add Golang SQLite test for CI

### PR DESCRIPTION
**Pull request by james-dong-security**
Enable Golang web server, Golang gRPC, and Golang SQLite tests for CI

### CODE REVIEW

**Review by jessehui — CHANGES_REQUESTED** [review]
Thank you for submitting this. Looks good in general.  Add some comments about the format.

File: .github/workflows/demo_test.yml [hunk 1]
```diff
@@ -332,3 +333,111 @@ jobs:
     # TODO: Remove " || true " because in simulation mode, process exits brutely and can cause segmentation fault in some cases.
     - name: Run xgboost cluster test
       run: docker exec xgboost_test bash -c "cd /root/occlum/demos/xgboost && SGX_MODE=SIM make test-local-cluster" || true
+
+
+  Golang_WebServer_test:
```
[the remark this answers is not in the document]
Comment (james-dong-security) [reply .github/workflows/demo_test.yml]: Thanks for pointing this out. Removed the duplicate runs.

File: .github/workflows/demo_test.yml [hunk 2]
```diff
@@ -332,3 +333,111 @@ jobs:
...
+        docker exec golang_grpc_test bash -c "cd /root/occlum/demos/golang/grpc_pingpong/ && SGX_MODE=SIM ./run_ping_on_occlum.sh" || true
+
+    # TODO: When sgx simulation mode supports interrupt signal, remove this.
+    - name: Kill the container
+      run: |
+        sleep ${{ env.nap_time }};
+        docker stop golang_grpc_test
+
+
+  Golang_SQLite_test:
```
[the remark this answers is not in the document]
Comment (james-dong-security) [reply .github/workflows/demo_test.yml]: Done

File: .github/workflows/demo_test.yml [hunk 3]
```diff
@@ -332,3 +333,111 @@ jobs:
...
+        docker exec golang_webserver_test bash -c "cd /root/occlum/demos/golang/web_server/ && curl http://127.0.0.1:8090/ping"
+
+    # TODO: When sgx simulation mode supports interrupt signal, remove this.
+    - name: Kill the container
+      run: |
+        sleep ${{ env.nap_time }};
+        docker stop golang_webserver_test
+
+
+  Golang_gRPC_test:
```
[the remark this answers is not in the document]
Comment (james-dong-security) [reply .github/workflows/demo_test.yml]: Thanks for pointing this out. Removed the duplicate runs.

**Comment by james-dong-security** [comment]
@jessehui : 
Thanks for reviewing earlier patch, and for catching the duplicate tests. Made changes accordingly. PTAL.

**Review by jessehui — APPROVED** [review]
LGTM

**Review by jessehui — APPROVED** [review]

**Review by tatetian — APPROVED** [review]
LGTM

**Pull request merged**